middleware.go

```diff
@@ -292,22 +292,35 @@ func EnableCORS(method string, path string) echo.MiddlewareFunc {
 	})
 }
 
-type WEOSError struct {
-	Message    string `json:"message"`
-	StatusCode int    `json:"statusCode"`
-	Error      error  `json:"error"`
+type WeOSControllerError struct {
+	message    string
+	statusCode int
+	err        error
+}
+
+func (e *WeOSControllerError) Error() string {
+	return e.message
+}
+
+func (e *WeOSControllerError) Unwrap() error {
+	return e.err
+}
+
+func NewControllerError(message string, err error, code int) *WeOSControllerError {
+	return &WeOSControllerError{
+		message:    message,
+		err:        err,
+		statusCode: code,
+	}
 }
 
 func customHTTPErrorHandler(err error, c echo.Context) {
 	code := http.StatusInternalServerError
 	if he, ok := err.(*echo.HTTPError); ok {
 		code = he.Code
 	}
-	weosError := &WEOSError{
-		StatusCode: code,
-		Error:      err,
-	}
-	if err := c.JSON(code, weosError); err != nil {
+	controllerError := NewControllerError("", err, code)
```

Comment:
I don't think you would be instantiating the error. The err that comes in would in theory be a ControllerError that you would check in here.

Reply:
You should check to see if the status code is 0 and if so set a default status code based on the error type that is wrapped within the controller error that the handler is receiving